toolchain/check/merge.cpp

```diff
@@ -6,11 +6,35 @@
 
 #include "toolchain/base/kind_switch.h"
 #include "toolchain/check/import_ref.h"
+#include "toolchain/diagnostics/format_providers.h"
 #include "toolchain/sem_ir/ids.h"
 #include "toolchain/sem_ir/typed_insts.h"
 
 namespace Carbon::Check {
 
+// Explicit or implicit parameters. Used in diagnostics.
+enum class ParamKind : uint8_t {
+  Explicit,
+  Implicit,
+};
+
+}  // namespace Carbon::Check
+
+// `ParamKind` is formatted in diagnostics as `{0}parameters`, and we only add
+// text for implicit paramseters.
+template <>
+struct llvm::format_provider<Carbon::Check::ParamKind> {
+  using ParamKind = Carbon::Check::ParamKind;
+  static void format(const ParamKind& kind, raw_ostream& out,
+                     StringRef /*style*/) {
+    if (kind == ParamKind::Implicit) {
+      out << "implicit ";
+    }
+  }
+};
```

Comment:
This doesn't look like it would support things like diagnostic translation well; could we use something like `FormatBool` instead?